src/lib/molecules/InteractiveWidget/InteractiveWidget.tsx

```diff
@@ -0,0 +1,187 @@
+    /**
+     * Fires an event on Panel click `((event: Event) => void)`.
+     */
+    onClick?: (event: MouseEvent | KeyboardEvent) => void;
+    /**
+     * Disabled state.
+     */
+    disabled?: boolean;
+    /**
+     * The `icon` prop determines the representation of an icon within the component.
+     * It can be a string, such as a URL for an SVG image, or a React component for example <Icon> component.
+     */
+    icon?: string | ReactNode;
+    /**
+     * This property controls the size of the icon, influencing the appearance of the component.<br>
+     * Possible values: `'default' | 'compact'`
+     */
+    appearance?: 'default' | 'compact';
+    /**
+     * The `iconColor` prop determines the color of the icon in the component.
+     * It should be a string representing a valid CSS color value.
+     */
+    iconColor?: string;
+    /**
+     * Label for tag component.
+     * Adding props you will add the `<Tag/>` component.
+     */
+    tagName?: string;
+    /**
+     * color for tag.
+     */
+    tagColor?: string;
+    /**
+     * Props for `<Switcher/>` component (see `<Switcher/>` component documentation).
+     * Adding props you wil add the component.
+     */
+    switcherProps?: typeof Switcher.propTypes;
+    /**
+     * The main descriptive text or title for the component.
+     */
+    title?: string;
+    /**
+     * Additional information or tooltip string that appears when hovering over the info icon next to the title.
+     * If not provided, the info icon will not be displayed.
+     */
+    titleInfo?: string;
+    /**
+     * Description paragraph.
+     */
+    description?: string;
+    /**
+     * icon background for default appearance.
+     */
+    iconBackground?: boolean;
+}
+
+const InteractiveWidget: FC<IInteractiveWidgetProps> = ({
+    className,
+    withBorder = true,
+    onClick,
+    disabled = false,
+    icon,
+    title,
+    appearance = 'default',
+    iconColor,
+    tagName,
+    tagColor,
+    switcherProps,
+    titleInfo,
+    description,
+    iconBackground = true,
+    ...restProps
+}) => {
+    const titleRef = useRef<HTMLHeadingElement | null>(null);
+    const descriptionRef = useRef<HTMLParagraphElement | null>(null);
+    const isTitleTruncated = useEllipsisDetection(titleRef);
+    const isDescriptionTruncated = useEllipsisDetection(descriptionRef);
+
+    return (
+        <div
+            className={classNames(
+                'interactiveWidget',
+                {
+                    'interactiveWidget-border': withBorder,
+                    'interactiveWidget-pointer': onClick,
+                    'interactiveWidget-disabled': disabled
+                },
+                className
+            )}
+            {...(!disabled && onClick ? { onClick: (event: MouseEvent<HTMLDivElement>) => onClick(event) } : {})}
+            tabIndex={0}
+            onKeyDown={(event: KeyboardEvent<HTMLDivElement>) => {
+                onClick && onClick(event);
```

Comment:
Here you need to add check and `return` from the event handler in case the `disabled` prop is `true` as the `pointer-events: none;` does not prevent the keyboard events. So you will have a bug here.

```suggestion
                if (disabled) return;
                onClick && onClick(event);
```